ddtrace/internal/packages.py

```diff
@@ -0,0 +1,30 @@
+import typing
+
+
+Distribution = typing.NamedTuple("Distribution", [("name", str), ("version", str)])
+
+_DISTRIBUTIONS = None  # type: typing.Optional[typing.Set[Distribution]]
+
+
+def get_distributions():
+    # type: () -> typing.Set[Distribution]
+    global _DISTRIBUTIONS
+    if _DISTRIBUTIONS is not None:
+        return _DISTRIBUTIONS
+
+    try:
+        import importlib.metadata as importlib_metadata
+    except ImportError:
+        import importlib_metadata  # type: ignore[no-redef]
+
+    pkgs = set()
+    for dist in importlib_metadata.distributions():
+        metadata = dist.metadata
+        name = metadata["name"]
+        version = metadata["version"]
+        if name and version:
+            # add distribution to a set to remove duplicates
```

Comment:
not 100% true, since we might get duplicate packages but different versions.

can we clarify that information here? Or maybe better as a docstring to describe which packages this will pull, the fact that we might pull "duplicates" if the versions are different, etc

Reply:
A duplicate package name and version maybe returned by `importlib_metadata.distributions()`. For example in a case where the base riot venv has a package that is also installed in venv of a contrib test. I'll make this more descriptive in the doc string

Reply:
yeah, thanks. the way this is written "remove duplicates", it doesn't really capture that we can have duplicate package names, but not duplicate package name + version entries.